pallets/airdrop/src/vested_transfer.rs

```diff
@@ -0,0 +1,125 @@
+use crate as airdrop;
+use airdrop::{error, info};
+use airdrop::{types, utils, Pallet as AirdropModule};
+use frame_support::pallet_prelude::*;
+use frame_support::traits::{Currency, ExistenceRequirement};
+use sp_runtime::traits::{CheckedAdd, Convert};
+
+// TODO: put more relaible value
+pub const BLOCKS_IN_YEAR: u32 = 5_256_000u32;
+// Block number after which enable to do vesting
+pub const VESTING_APPLICABLE_FROM: u32 = 1u32;
+
+pub struct DoVestdTransfer;
+impl types::DoTransfer for DoVestdTransfer {
+	fn do_transfer<T: airdrop::Config>(snapshot: &mut types::SnapshotInfo<T>) -> DispatchResult {
+		let vesting_should_end_in = <T as airdrop::Config>::AIRDROP_VARIABLES.vesting_period;
+		let defi_user = snapshot.defi_user;
+		let total_amount = snapshot.amount;
+
+		let claimer = &snapshot.ice_address;
+		let creditor = AirdropModule::<T>::get_creditor_account()?;
+
+		let instant_percentage = utils::get_instant_percentage::<T>(defi_user);
+		let (mut instant_amount, vesting_amount) =
+			utils::get_splitted_amounts::<T>(total_amount, instant_percentage).map_err(|e |{
+				error!("At: get_splitted_amount. amount: {total_amount:?}. Instant percentage: {instant_percentage}. Reason: {e:?}");
+				e
+			})?;
+
+		let (transfer_shcedule, remainding_amount) = utils::new_vesting_with_deadline::<
+			T,
+			VESTING_APPLICABLE_FROM,
+		>(vesting_amount, vesting_should_end_in.into());
+
+		// Amount to be transferred is:
+		// x% of totoal amount
+		// + remainding amount which was not perfectly divisible
+		instant_amount = {
+			let remainding_amount = <T::BalanceTypeConversion as Convert<
+				types::VestingBalanceOf<T>,
+				types::BalanceOf<T>,
+			>>::convert(remainding_amount);
+
+			instant_amount
+				.checked_add(&remainding_amount)
+				.ok_or(sp_runtime::ArithmeticError::Overflow)?
+		};
+
+		let creditor_origin = <T as frame_system::Config>::Origin::from(
+			frame_system::RawOrigin::Signed(creditor.clone()),
+		);
+		let claimer_origin =
+			<T::Lookup as sp_runtime::traits::StaticLookup>::unlookup(claimer.clone());
+
+		match transfer_shcedule {
+			// Apply vesting
+			Some(schedule) if !snapshot.done_vesting => {
+				let vest_res = pallet_vesting::Pallet::<T>::vested_transfer(
+					creditor_origin,
+					claimer_origin,
+					schedule,
+				);
+
+				match vest_res {
+					// Everything went ok. update flag
+					Ok(()) => {
+						let block_number = utils::get_current_block_number::<T>();
+						snapshot.done_vesting = true;
+						snapshot.vesting_block_number = Some(block_number);
+
+						info!("Vesting applied for {claimer:?} at height {block_number:?}");
+					}
+					// log error
+					Err(err) => {
+						info!("Error while aplying vesting. For: {claimer:?}. Reason: {err:?}");
```

Comment:
can use use `error!` instead of `info!` when logging errors?

Reply:
Originally, I thought maybe we should separate node error from pallet error. But looks like using `error!` will be better. This will be in-progress from our side